spark/src/main/java/org/apache/iceberg/spark/hacks/Hive.java

```diff
@@ -62,4 +63,16 @@ private Hive() {}
 
     return client.getPartitionsByFilter(client.getTable(db, table), exprs);
   }
+
+  public static CatalogTable getTable(SparkSession spark, String name) {
+    List<String> parts = Lists.newArrayList(Splitter.on('.').limit(2).split(name));
+    String db = parts.size() == 1 ? "default" : parts.get(0);
+    String table = parts.get(parts.size() == 1 ? 0 : 1);
+
+    HiveClient client = HiveUtils$.MODULE$.newClientForMetadata(
```

Comment:
Is there a specific reason to use `HiveClient` directly? Would it make sense to use `spark.sessionState.catalog`? That way, we will also support the in-memory catalog and any other catalog that can return `CatalogTable` for us.

Reply:
Agree to use Spark's session catalog rather than directly manipulating hive or other catalog implementations.

Reply:
OK, updated.